src/main/kotlin/org/rust/ide/folding/RustFoldingBuilder.kt

```diff
@@ -0,0 +1,85 @@
+package org.rust.ide.folding
+
+import com.intellij.lang.ASTNode
+import com.intellij.lang.folding.FoldingBuilderEx
+import com.intellij.lang.folding.FoldingDescriptor
+import com.intellij.openapi.editor.Document
+import com.intellij.openapi.util.TextRange
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.impl.RustFile
+import java.util.*
+
+class RustFoldingBuilder() : FoldingBuilderEx() {
+    override fun getPlaceholderText(node: ASTNode): String? {
```

Comment:
Can be more concisely expressed as `override fun getPlaceholderText(node: ASTNode): String = "{...}" `